phoenix-core/src/main/java/org/apache/phoenix/expression/LiteralExpression.java

```diff
@@ -57,181 +56,236 @@
 	private static final LiteralExpression[] NULL_EXPRESSIONS = new LiteralExpression[Determinism.values().length];
     private static final LiteralExpression[] TYPED_NULL_EXPRESSIONS = new LiteralExpression[PDataType.values().length * Determinism.values().length];
     private static final LiteralExpression[] BOOLEAN_EXPRESSIONS = new LiteralExpression[2 * Determinism.values().length];
-    
+
     static {
     	for (Determinism determinism : Determinism.values()) {
-    		NULL_EXPRESSIONS[determinism.ordinal()] = new LiteralExpression(null, determinism);
+            NULL_EXPRESSIONS[determinism.ordinal()] = new LiteralExpression(null, null,
```

Comment:
For this whole static block, I decided to call the constructor directly since the original code is calling the constructor directly.
Additionally, using a builder here would necessitate creating a new build method (or adding another path to my existing build method) which I decided not to do since it would only be used twice throughout the code